source/imaer-gml/src/test/java/nl/overheid/aerius/test/TestValidationAndEmissionHelper.java

```diff
@@ -637,29 +639,32 @@ public boolean isValidRoadSpecificVehicleCode(final String specificVehicleCode)
   }
 
   @Override
-  public boolean isValidRoadStandardVehicleCombination(final String roadAreaCode, final String roadTypeCode, final String vehicleTypeCode,
-      final Integer maximumSpeed,
-      final Boolean strictEnforced) {
-    return roadStandard(vehicleTypeCode, roadTypeCode, maximumSpeed, strictEnforced).isPresent();
+  public boolean isValidRoadStandardVehicleCombination(final RoadStandardEmissionFactorsKey emissionFactorsKey) {
+    return roadStandard(emissionFactorsKey).isPresent();
   }
 
   @Override
-  public Map<Substance, Double> getRoadStandardVehicleEmissionFactors(final String roadAreaCode, final String standardVehicleCode,
-      final String roadTypeCode,
-      final Integer maximumSpeed, final Boolean strictEnforcement) {
-    return roadStandard(standardVehicleCode, roadTypeCode, maximumSpeed, strictEnforcement)
+  public Map<Substance, Double> getRoadStandardVehicleEmissionFactors(final RoadStandardEmissionFactorsKey emissionFactorsKey) {
+    return roadStandard(emissionFactorsKey)
         .map(c -> c.emissions.toEmissions())
         .orElseThrow();
   }
 
   @Override
-  public Map<Substance, Double> getRoadStandardVehicleStagnatedEmissionFactors(final String roadAreaCode, final String standardVehicleCode,
-      final String roadTypeCode, final Integer maximumSpeed, final Boolean strictEnforcement) {
-    return roadStandard(standardVehicleCode, roadTypeCode, maximumSpeed, strictEnforcement)
+  public Map<Substance, Double> getRoadStandardVehicleStagnatedEmissionFactors(final RoadStandardEmissionFactorsKey emissionFactorsKey) {
+    return roadStandard(emissionFactorsKey)
         .map(c -> c.stagnatedEmissions.toEmissions())
         .orElseThrow();
   }
 
+  @Override
+  public RoadStandardsInterpolationValues getRoadStandardVehicleInterpolationValues(final RoadStandardEmissionFactorsKey emissionFactorsKey) {
+    // For now just return the supplied values (will result in no interpolation).
+    final int speed = emissionFactorsKey.getMaximumSpeed();
+    final int gradient = Optional.ofNullable(emissionFactorsKey.getGradient()).map(Double::intValue).orElse(0);
```

Comment:
This will round the double down if I'm correct. Is that intended or a problem?

Reply:
For testing purposes it shouldn't matter as we don't really have gradient test values or cases (yet). Might change this later on when working on the gradient story.